src/Microsoft.Tye.Hosting/Infrastructure/ProxyExtensions.cs

```diff
@@ -85,6 +85,11 @@ public static HttpRequestMessage CreateProxyHttpRequest(this HttpContext context
                 }
             }
 
+            // Append request forwarding headers
+            requestMessage.Headers.TryAddWithoutValidation("X-Forwarded-For", context.Connection.LocalIpAddress.ToString());
```

Comment:
@Tratcher does YARP do anything by default for adding the x-forwarded headers?

Also, websockets. What would we need to add to make websockets work?

Reply:
Yes, YARP adds x-forwarded headers by default.
https://microsoft.github.io/reverse-proxy/articles/transforms.html#defaults

x-forwarded headers should be applied equally to websocket requests.

Careful, I don't think TryAddWithoutValidation actually appends to existing values. Test it out.

Reply:
We always want to append here, correct?

@FiniteReality could you validate what happens when TryAddWithoutValidation is called on an already existing value? And also look into how these could apply to websockets.

Reply:
Looking at [the source](https://source.dot.net/#System.Net.Http/System/Net/Http/Headers/HttpHeaders.cs,e7ab97274f238802), TryAddWithoutValidation appears to append to existing values if one is present.

I think applying these to WebSockets should be relatively straightfoward too - I'll look into it later today.